test/support/factory.ex

```diff
@@ -272,6 +272,13 @@ defmodule Oli.Factory do
     %Resource{}
   end
 
+  def project_resource_factory() do
+    %ProjectResource{
+      project_id: insert(:project).id,
```

Comment:
Super nit but you can also do
```elixir 
project: insert(:project)
```

Reply:
Makes sense! Copied a bad habit from the schema 😅

Reply:
I couldn't manage to do this, the ProjectResource schema is defined as such: 
```
  schema "projects_resources" do
    timestamps(type: :utc_datetime)
    field :project_id, :integer, primary_key: true
    field :resource_id, :integer, primary_key: true
  end
```
Therefore the `ex_machina` factory needs to be defined as:

```
 def project_resource_factory() do
    %ProjectResource{
      project_id: insert(:project).id,
      resource_id: insert(:resource).id
    }
  end
```

And it seems like `ex_machina` doesn't allow for receiving the whole struct and just taking the id.